fix(parser): ensure dateRange uses startDate for condition

pmendozav opened the pull request:
related with issue https://github.com/videojs/m3u8-parser/issues/186

pmendozav committed:
fix(parser): ensure dateRange uses startDate for condition

aromansh commented on src/parser.js:
```diff
@@ -715,7 +715,7 @@ export default class Parser extends Stream {
                 });
               }
-              if (dateRange.duration && dateRange.endDate) {
+              if (dateRange.duration && dateRange.startDate) {
```
But we also use `endDate` in line 722, we should check that value too

ishish commented:
Is there an update on this can we get it merged?

adrums86 commented on src/parser.js:
```diff
@@ -715,7 +715,7 @@ export default class Parser extends Stream {
                   message: 'EXT-X-DATERANGE with an END-ON-NEXT=YES attribute must not contain DURATION or END-DATE attributes'
                 });
               }
-              if (dateRange.duration && dateRange.endDate) {
```
I'll defer to the discussion in #186 , but I think we may want to do something like
```js
if (dateRange.startDate && dateRange.duration && dateRange.endDate)
```
instead. 
Perhaps with a warning and early return if the `startDate + duration !== endDate` to ensure we don't create any bad dateRanges.

pmendozav replied:
@adrums86 

I updated the constraint to include `dateRange.endDate`, ensuring that `dateRanges[index].endDate` is created according to the specification and that a valid `startDate` is present. This should be sufficient to prevent the creation of invalid `dateRanges`, as you mentioned.

Additionally, I added a warning.

adrums86 reviewed (changes requested):
Thanks for submitting a fix! I agree this could be handled better. I think we might just want to be careful about possibly creating out of spec dateRange tags so lets see if we can do this as close to spec as possible.

pmendozav committed:
chore(logging): START-DATE missing in EXT-X-DATERANGE

Triggers a warn event when an EXT-X-DATERANGE tag includes both DURATION and END-DATE but lacks the required START-DATE. Ensures better HLS spec compliance.

gkatsev reviewed (approved):
tests pass locally

adrums86 reviewed (approved):
:shipit: Obviously some karma issues that need fixed in the repo. Seems to pass locally, so its unrelated.

pmendozav commented:
Any update here? Can the change be merged?
